torch/csrc/jit/tensorexpr/bounds_inference.cpp

```diff
@@ -78,6 +78,16 @@ std::unordered_map<const Var*, const Buf*> getAllBufs(Stmt* s) {
   return varToBuf;
 }
 
+std::unordered_map<const Var*, const Buf*> getAllBufs(Expr* e) {
```

Comment:
It's kind of unfortunate to have these duplicated functions for Stmt and Expr.  Maybe use a function template instead like
```
template<typename T>
std::unordered_map<const Var*, const Buf*> getAllBufs(T e) {
```

(At some point did someone want to unify `Expr` and `Stmt`?  I wonder if that's a good idea...)

Reply:
Interesting prior art from clang: https://clang.llvm.org/doxygen/classclang_1_1Expr.html#details

Reply:
> At some point did someone want to unify Expr and Stmt? 

Was this considered before? IMO, that would be great for the IR.

Reply:
Yeah, I like that idea too!